hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,235 @@
+     * Registers all client statistics and schedules peridic collection of stats
+     */
+    public void start() {
+        if (!enabled) {
+            return;
+        }
+
+        long periodSeconds = properties.getSeconds(PERIOD_SECONDS);
+        if (periodSeconds < 0) {
+            long defaultValue = Long.parseLong(PERIOD_SECONDS.getDefaultValue());
+            logger.warning(
+                    "Provided client statistics " + PERIOD_SECONDS.getName() + " can not be a negative number. You provided "
+                            + periodSeconds + ". Client will use the default value of " + defaultValue + " instead.");
+            periodSeconds = defaultValue;
+        }
+
+        staticStats.put("/enterprise", Boolean.toString(enterprise));
+        // Register for one time statistics. These are resent only when reconnected to the cluster.
+        registerStaticStatistics();
+
+        registerPeriodicMetrics();
+
+        logger.finest("Client statistics is enabled with period " + periodSeconds + " seconds.");
+
+        // Start task for sending periodic statistics
+        scheduleStatisticsCollection(periodSeconds);
+    }
+
+    private void scheduleStatisticsCollection(long periodSeconds) {
+        client.getExecutionService().scheduleWithRepetition(new Runnable() {
+            @Override
+            public void run() {
+                final Map<String, String> stats = new HashMap<String, String>();
+
+                metricsRegistry.render(new ProbeRenderer() {
+                    @Override
+                    public void renderLong(String name, long value) {
+                        stats.put(composeKey(name), Long.toString(value));
+                    }
+
+                    @Override
+                    public void renderDouble(String name, double value) {
+                        stats.put(composeKey(name), Double.toString(value));
+                    }
+
+                    /**
+                     * Converts keys like runtime.freeMemory to /runtime/freeMemory
+                     * @param name The name of the statistics
+                     * @return The statistics key name as properly formatted
+                     */
+                    private String composeKey(String name) {
+                        return "/" + name.replace('.', '/');
+                    }
+
+                    @Override
+                    public void renderException(String name, Exception e) {
+                        // No intrusive message for client statistics is desired
+                        logger.finest("Exception while processing statistics " + name + ". Error:" + e.getMessage());
+                    }
+
+                    @Override
+                    public void renderNoValue(String name) {
+                    }
+                });
+
+                // put user executor queue size
+                ThreadPoolExecutor userExecutor = (ThreadPoolExecutor) client.getClientExecutionService().getUserExecutor();
```

Comment:
Yes, I agree, we can add this to Diagnostics. It will be better to send a separate PR for this change, since changing the Diagnostics related statistics is not in the scope of this PR.